tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -1932,6 +1970,76 @@ public:
    ~RField() override = default;
 };
 
+template <typename ItemT>
+class RField<std::set<ItemT>> : public RSetField {
+   using ContainerT = typename std::set<ItemT>;
+
+protected:
+   std::size_t AppendImpl(const Detail::RFieldValue &value) final
+   {
+      auto typedValue = value.Get<ContainerT>();
+      auto nbytes = 0;
+      auto count = typedValue->size();
+      for (auto item : *typedValue) {
+         auto itemValue = fSubFields[0]->CaptureValue(static_cast<void *>(&item));
+         nbytes += fSubFields[0]->Append(itemValue);
+      }
+      Detail::RColumnElement<ClusterSize_t, EColumnType::kUnknown> elemIndex(&this->fNWritten);
+      this->fNWritten += count;
+      fColumns[0]->Append(elemIndex);
+      return nbytes + sizeof(elemIndex);
+   }
+
+   void ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value) final
+   {
+      auto typedValue = value->Get<ContainerT>();
+      RClusterIndex collectionStart;
+      ClusterSize_t nItems;
+      fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nItems);
+      typedValue->clear();
+      for (unsigned i = 0; i < nItems; ++i) {
+         ItemT item;
```

Comment:
```suggestion
      ItemT item;
      for (unsigned i = 0; i < nItems; ++i) {
```
Sometimes the item is 'expensive to default construct.  The I/O is usually resilient to reuse of the object to read into it multiple time.